docs/patterns/sqlite3.rst

```diff
@@ -71,18 +71,36 @@ Now in each request handling function you can access `g.db` to get the
 current open database connection.  To simplify working with SQLite, a
 row factory function is useful.  It is executed for every result returned
 from the database to convert the result.  For instance, in order to get
-dictionaries instead of tuples, this could be inserted into ``get_db``::
+dictionaries instead of tuples, this could be inserted into the ``get_db`` 
+function we created above::
 
     def make_dicts(cursor, row):
         return dict((cursor.description[idx][0], value)
                     for idx, value in enumerate(row))
 
     db.row_factory = make_dicts
 
-Or even simpler::
+This will make the sqlite3 module return dicts for this database connection, which are much nicer to deal with. Even more simply, we could place this in ``get_db`` instead::
 
     db.row_factory = sqlite3.Row
 
+This would use Row objects rather than dicts to return the results of queries. These are `namedtuple` s, so we can access them either by index or by key. For example, assuming we have a ``sqlite3.Row`` called ``r`` for the rows id, FirstName, LastName, and MiddleInitial::
```

Comment:
I think `namedtuples` needs two `` since it's restructured text and not markdown